scripts/build-macos-bundle.sh

```
@@ -0,0 +1,114 @@
+rm -rf "${BUNDLE_DIR}"
+mkdir -p "${BUNDLE_DIR}/lib"
+
+# ---- build gateway ------------------------------------------------------
+
+# GOFLAGS_STR is passed as a single string to preserve the `-ldflags "..."`
+# quoting; eval unpacks it in the go build invocation.
+build_arch() {
+  local arch="$1" out="$2"
+  echo "==> building gateway (darwin/${arch})"
+  eval GOOS=darwin GOARCH="${arch}" CGO_ENABLED=0 \
+    go build ${GOFLAGS_STR} -o "\"${out}\"" ./cmd/defenseclaw
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)